### PR TITLE
Improve error message when calling `inject()`.

### LINKED ISSUE #15904: Confusing message when using wrong `inject`
After upgrading to the new modules syntax, I had to revert one small section of my app (complicated story; mostly irrelevant), and in one module I missed converting `inject()` back to `inject.service()`. The error message reported is… extremely confusing:

> "Assertion Failed: Injected properties must be created through helpers, see 'controller"service'"

I managed to figure it out based on the fact that I knew I'd messed with rewriting injections in those particular modules, but I can imagine that being *especially* confusing if you didn't. (Note that the codemod eliminates this problem by doing `import { inject as service } from '@ember/service'`, so the error would be different—and clearer—in that case.

This error message should be much clearer.

### PR DESCRIPTION
**Pull request by chriskrycho**
Fixes #15904

### CODE REVIEW

File: packages/ember-runtime/lib/inject.js [hunk 1]
```diff
@@ -10,7 +10,10 @@ import { assert } from 'ember-debug';
   @public
 */
 export default function inject() {
-  assert(`Injected properties must be created through helpers, see '${Object.keys(inject).join('"', '"')}'`);
+  const helpers = Object.keys(inject)
```
Comment (bekzod) [inline packages/ember-runtime/lib/inject.js]: I think this should inlined inside `assert` else it won't be stripped

Comment (chriskrycho) [reply packages/ember-runtime/lib/inject.js]: I wondered about that, and can push an update to that effect tomorrow.

Comment (Serabe) [reply packages/ember-runtime/lib/inject.js]: That would solve the linting problem as well. Thank you!

Comment (chriskrycho) [reply packages/ember-runtime/lib/inject.js]: Updated!

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**